app/core/core-ui/src/main/kotlin/com/hedvig/android/core/ui/layout/LayoutWithoutPlacement.kt

```diff
@@ -0,0 +1,30 @@
+package com.hedvig.android.core.ui.layout
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.layout
+
+/**
+ * A [Layout] which places [sizeAdjustingContent] first in a box, making that box take *at least* its size but does not
+ * draw [sizeAdjustingContent] on the screen, and then puts [content]
+ */
+@Composable
+fun LayoutWithoutPlacement(
+  modifier: Modifier = Modifier,
+  sizeAdjustingContent: @Composable BoxScope.() -> Unit,
```

Comment:
So we need this for aligning according to the user's custom text size, right? Are there other scenarios where this could be adjusting to something other than text size? I was just thinking that if it's only text in all scenarios, we could probably change it to smth like ```fun LayoutCenteredAdjustedToTextSize``` and pass just textSize size there instead of sizeAdjustingContent, would be a bit easier to understand its purpose

Reply:
Yeah the name of this composable is also kinda horrible, I didn't find something better.
Our current use case is in fact only texts.
I did not do it like that because then it becomes impossible to adjust the text's font size, which is a very important thing to do from some callers. And if I expose the font size as a parameter too, then as we use it more we will simply keep adding parameters here only to forward them to the text.
The composable slot avoids any of this problem all together. I think I may have shared this before https://chrisbanes.me/posts/slotting-in-with-compose-ui/ but it really is a great way to introduce flexibility to such layouts.